ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerBuilder.java

```diff
@@ -66,9 +66,20 @@
   private @Nullable Map<String, RequestHandler> mCustomPackagerCommandHandlers;
   private @Nullable ReactPackageTurboModuleManagerDelegate.Builder mTMMDelegateBuilder;
   private @Nullable SurfaceDelegateFactory mSurfaceDelegateFactory;
+  private String jscOrHermes = "";
```

Comment:
Why are you using an extra string here? A boolena would be sufficient

Reply:
I am using it to keep the old logic intact.
In this way, app already using the old logic won't need to change anything and those who were facing crashes can update. The empty case uses the old try catch logic.

Reply:
It can be updated to use a boolean only if you need to mandate all apps to use the new approach and pass the value for hermesEnabled in the ReactInstanceManager :: buidler method.
The old logic would be removed in that case.

Reply:
Let's update this to be a `Boolean hermesEnabled = null`, with the current logic:

- `null` -> Use the previous logic
- `true` -> Load Hermes or fail
- `false` -> Load JSC or fail